app/dao/fact_billing_dao.py

```diff
@@ -19,11 +24,155 @@
+                     ], else_=0)
+
+    query = db.session.query(
+        Organisation.name.label('organisation_name'),
+        Organisation.id.label('organisation_id'),
+        Service.name.label("service_name"),
+        FactBilling.service_id.label("service_id"),
+        AnnualBilling.free_sms_fragment_limit,
+        FactBilling.rate.label('sms_rate'),
+        sms_remainder.label("sms_remainder"),
+        sms_billable_units.label('sms_billable_units'),
+        chargeable_sms.label("chargeable_billable_sms"),
+        sms_cost.label('sms_cost'),
+    ).join(
+        Service.annual_billing,
+    ).outerjoin(
+        free_allowance_remainder, Service.id == free_allowance_remainder.c.service_id
+    ).outerjoin(
+        Organisation, Service.organisation_id == Organisation.id
+    ).filter(
+        FactBilling.service_id == Service.id,
+        FactBilling.bst_date >= start_date,
+        FactBilling.bst_date <= end_date,
+        FactBilling.notification_type == SMS_TYPE,
+        AnnualBilling.financial_year_start == billing_year,
+    ).group_by(
+        Organisation.name,
+        Organisation.id,
+        FactBilling.service_id,
+        Service.name,
+        AnnualBilling.free_sms_fragment_limit,
+        free_allowance_remainder.c.sms_remainder,
+        FactBilling.rate,
+    ).order_by(
+        Organisation.name,
+        Service.name
+    )
+
+    return query.all()
+
+
+def fetch_letter_costs_for_all_services(start_date, end_date):
+    query = db.session.query(
+        Organisation.name.label("organisation_name"),
+        Organisation.id.label("organisation_id"),
+        Service.name.label("service_name"),
+        FactBilling.service_id.label("service_id"),
+        func.sum(FactBilling.notifications_sent * FactBilling.rate).label("letter_cost")
+    ).outerjoin(
+        Organisation, Service.organisation_id == Organisation.id
+    ).filter(
+        FactBilling.service_id == Service.id,
+        FactBilling.bst_date >= start_date,
+        FactBilling.bst_date <= end_date,
+        FactBilling.notification_type == LETTER_TYPE,
+    ).group_by(
+        Organisation.name,
+        Organisation.id,
+        FactBilling.service_id,
+        Service.name,
+    ).order_by(
+        Organisation.name,
+        Service.name
+    )
+
+    return query.all()
+
+
+def fetch_letter_line_items_for_all_services(start_date, end_date):
+    query = db.session.query(
+        Organisation.name.label("organisation_name"),
+        Organisation.id.label("organisation_id"),
+        Service.name.label("service_name"),
+        FactBilling.service_id.label("service_id"),
+        FactBilling.billable_units.label("sheet_count"),
+        FactBilling.rate.label("letter_rate"),
+        FactBilling.postage.label("postage"),
+        func.sum(FactBilling.notifications_sent).label("letters_sent"),
+    ).outerjoin(
+        Organisation, Service.organisation_id == Organisation.id
```

Comment:
similar join comment. This table uses Organisation, Service and FactBilling so those should be joined.